internal/controller/controllers/agent_controller.go

```diff
@@ -154,13 +154,6 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 
 		// Retrieve clusterDeployment
 		if err = r.Get(ctx, kubeKey, clusterDeployment); err != nil {
-			if k8serrors.IsNotFound(err) {
-				// Delete the agent, using a finalizer with pre-delete to deregister the host.
-				log.Infof("Cluster Deployment name: %s namespace: %s not found, deleting Agent",
-					agent.Spec.ClusterDeploymentName.Name, agent.Spec.ClusterDeploymentName.Namespace)
-				return r.deleteAgent(ctx, log, req.NamespacedName)
```

Comment:
Are agents for non-existing clusters not going to be deleted?

Reply:
Maybe we need to change the condition to delete them if infra env is deleted.

Reply:
There is another Jira ticket for this flow : https://issues.redhat.com/browse/MGMT-7758